src/server/api/routes/users/v2/email.js

```diff
@@ -0,0 +1,118 @@
+const { Router } = require('express');
+const { v4: uuidv4 } = require('uuid');
+const {
+  db,
+  FieldValue,
+  walletUserCollection,
+} = require('../../../../modules/firebase');
+const {
+  authenticateV2Login,
+  checkWalletMatch,
+} = require('../../../middleware/auth');
+const { handleRestfulError } = require('../../../middleware/auth');
+const { sendEmail } = require('../../../../modules/sendgrid');
+
+const {
+  VERIFICATION_EMAIL_RESEND_COOLDOWN_IN_MS,
+} = require('../../../constant');
+const { EXTERNAL_URL } = require('../../../../config/config');
+
+const router = Router();
+
+router.post(
+  '/v2/users/:wallet/email',
+  authenticateV2Login,
+  checkWalletMatch,
```

Comment:
How can frontend know email of current login user, and its verification status? in `/self`?

Reply:
Yes, also consider add followers/followees in `/self`?

Reply:
we might need followers in public api, so it depends
email should be in self though